src/commands/register-rab-bundle.ts

```diff
@@ -1,22 +1,31 @@
 /**
- * Copyright © 2023, Oracle and/or its affiliates.
+ * Copyright © 2022-2024, Oracle and/or its affiliates.
  * This software is licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl.
  */
 
-import * as fs from 'fs';
 import * as vscode from 'vscode';
 
-import * as _ from 'lodash';
 
+import { firstValueFrom } from 'rxjs';
 import * as api from '../api';
 import { log } from '../logger';
-import { createRABBundle } from '../workspace-manager';
 import { get as getProfileManager } from '../profile-manager-provider';
-import { RABError, showInfoMessage, showErrorMessage, withProgress } from '../utils/ui-utils';
+import { fs } from '../utils';
+import { RABError, showErrorMessage, showInfoMessage, withProgress } from '../utils/ui-utils';
+import { createRABBundle, getAddFile } from '../workspace-manager';
 
 async function callback(file: vscode.Uri, context: vscode.ExtensionContext): Promise<any> {
-
+  
   withProgress("Registering RAB bundle...", async () => {
+
+    const add = getAddFile(undefined, undefined, true);
+    if (!add) {
+      return;
+    }
+    const isSaved = await firstValueFrom(fs.confirmSaveFile(add, true));
```

Comment:
saved file check is already done in bundle validation method.

Reply:
Same as above.